core/src/main/scala/sttp/openai/OpenAISyncClient.scala

```diff
@@ -0,0 +1,447 @@
+    *
+    * @param embeddingsBody
+    *   Embeddings request body.
+    */
+  def createEmbeddings(embeddingsBody: EmbeddingsBody): Either[OpenAIError, EmbeddingResponse] =
+    sendAndHandleResponse(openAI.createEmbeddings(embeddingsBody))
+
+  /** Gets info about the fine-tune job.
+    *
+    * [[https://platform.openai.com/docs/api-reference/fine-tunes/retrieve]]
+    *
+    * @param fineTuneId
+    *   The ID of the fine-tune job.
+    */
+  def retrieveFineTune(fineTuneId: String): Either[OpenAIError, FineTuneResponse] =
+    sendAndHandleResponse(openAI.retrieveFineTune(fineTuneId))
+
+  /** Delete a fine-tuned model. You must have the Owner role in your organization.
+    *
+    * [[https://platform.openai.com/docs/api-reference/fine-tunes/delete-model]]
+    *
+    * @param model
+    *   The model to delete.
+    */
+  def deleteFineTuneModel(model: String): Either[OpenAIError, DeleteFineTuneModelResponse] =
+    sendAndHandleResponse(openAI.deleteFineTuneModel(model))
+
+  /** Get fine-grained status updates for a fine-tune job.
+    *
+    * [[https://platform.openai.com/docs/api-reference/fine-tunes/events]]
+    *
+    * @param fineTuneId
+    *   The ID of the fine-tune job to get events for.
+    */
+  def getFineTuneEvents(fineTuneId: String): Either[OpenAIError, FineTuneEventsResponse] =
+    sendAndHandleResponse(openAI.getFineTuneEvents(fineTuneId))
+
+  def close(): Unit = backend.close()
+
+  private def sendAndHandleResponse[A](
+      request: Request[Either[ResponseException[String, Exception], A]]
+  ): Either[OpenAIError, A] =
+    request.send(backend).body.left.map {
+      case e @ HttpError(body, statusCode) =>
+        import sttp.model.StatusCode._
+        val errorMessageBody = SnakePickle.read[ujson.Value](body).apply("error").obj.values
+        val (message, typ, param, code) =
+          SnakePickle.read[(Option[String], Option[String], Option[String], Option[String])](errorMessageBody)
+        statusCode match {
+          case TooManyRequests                              => RateLimitError(message, typ, param, code, e)
+          case BadRequest | NotFound | UnsupportedMediaType => InvalidRequestError(message, typ, param, code, e)
+          case Unauthorized                                 => AuthenticationError(message, typ, param, code, e)
+          case Forbidden                                    => PermissionError(message, typ, param, code, e)
+          case Conflict                                     => TryAgain(message, typ, param, code, e)
+          case ServiceUnavailable                           => ServiceUnavailableError(message, typ, param, code, e)
+          case _                                            => APIError(message, typ, param, code, e)
+        }
+      case DeserializationException(_, error) => throw new json.DeserializationException(error)
+    }
+}
+
+object OpenAISyncClient {
+  def apply(authToken: String) = new OpenAISyncClient(authToken, DefaultSyncBackend())
```

Comment:
here we also need to expose a `close` method (which closes the backend if it has been created when creating the client - see sttp, it has a similar mechanism)

Reply:
@adamw I created one in `class OpenAISyncClient`, but forgot to make docs for that (line 420):
```scala
def close(): Unit = backend.close()
```
I don't know if we need that in companion object tbh and if so how should that look like :)

Reply:
The problem is we should only close the backend if we opened it (if it has been provided externally, `close` should be a no-op). See here: https://github.com/softwaremill/sttp/blob/master/core/src/main/scalajvm/sttp/client4/httpclient/HttpClientSyncBackend.scala#L21